wsagent/che-core-api-project/src/test/java/org/eclipse/che/api/project/server/ProjectServiceTest.java

```diff
@@ -928,6 +939,50 @@ public void testDeleteProject() throws Exception {
     }
 
     @Test
+    public void testDeleteProjectsConcurrently() throws Exception {
+        int threadNumber = 100;
+        ExecutorService executor = Executors.newFixedThreadPool(threadNumber);
+        CountDownLatch countDownLatch = new CountDownLatch(threadNumber);
+        List<Future<ContainerResponse>> futures = new LinkedList<>();
+
+
+        for (int i = 0; i < threadNumber; i++) {
+            addMockedProjectConfigDto(ptRegistry.getProjectType("my_project_type"), "my_project_name" + i);
+        }
+
+        IntStream.range(0, threadNumber).forEach(
+                i -> {
+                    futures.add(executor.submit(() -> {
+                        countDownLatch.countDown();
+                        countDownLatch.await();
+
+                        try {
+                            return launcher.service(DELETE,
+                                                    "http://localhost:8080/api/project/my_project_name" + i,
+                                                    "http://localhost:8080/api", null, null, null);
+                        } catch (Exception e) {
+                            throw new IllegalStateException(e);
+                        }
+                    }));
+                }
+                                                );
+
+        boolean isNotDone;
+        do {
+            isNotDone = false;
+            for (Future<ContainerResponse> future : futures) {
+                if (!future.isDone()) {
+                    isNotDone = true;
+                }
+            }
+        } while (isNotDone);
```

Comment:
I think count down latch was designed exactly for such use cases. So +1 that it will be much clearer